src/main/play-25/uk/gov/hmrc/hmrcfrontend/config/ContactFrontendConfig.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hmrcfrontend.config
+
+import javax.inject.Inject
+import play.api.Configuration
+import play.api.mvc.RequestHeader
+
+class ContactFrontendConfig @Inject()(config: Configuration) {
+  private val platformHost: Option[String] =
+    config.getString("platform.frontend.host")
+
+  val baseUrla: Option[String] =
+    platformHost.orElse(config.getString("contact-frontend.host"))
+  val serviceCodea: Option[String] =
+    config.getString("contact-frontend.serviceCode")
+
+  def referrerUrl(implicit request: RequestHeader): Option[String] = {
+    val query = if (request.rawQueryString != "") { s"?${request.rawQueryString}" } else { "" }
```

Comment:
Should we be including the querystring? We opted not to do this in AccessibilityStatementConfig [here](https://github.com/hmrc/play-frontend-hmrc/blob/5108877836639f4af47449ebee207e65147840bd/src/main/play-26/uk/gov/hmrc/hmrcfrontend/config/AccessibilityStatementConfig.scala#L44) due to the risk of ending up with an invalid URI length.

Reply:
The query string is part of the URL, I don't see why we'd remove it.

Maximum URL lengths are pretty insane on most browsers & serverside tech.  Do we have a limit somewhere we're trying to avoid?

The old limit with Internet Explorer was a little over 2048, in Edge it's much more (e.g. 81578 reported here https://stackoverflow.com/questions/32267442/url-length-limitation-of-microsoft-edge#:~:text=2%20Answers&text=The%20limit%20appears%20to%20be%20around%2081578%20characters.)  If we have a URL of 1,000 characters anywhere on the platform a) we're probably doing something wrong and b) we're still nowhere near the limit.

Reply:
Contact frontend currently has a 2048 character limit on URLs and there have been instances of services linking to contact frontend receiving errors when passing on very long query string parameters. If we can remove the limitation from contact-frontend then it should be fine.